target/linux/ath79/dts/ar9331_arduino_yun.dts

```diff
@@ -0,0 +1,223 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+#include "ar9331.dtsi"
+
+/ {
+	model = "Arduino Yun";
+	compatible = "arduino,yun", "qca,ar9331";
+
+	aliases {
+		serial0 = &uart;
+	};
+
+	chosen {
+		bootargs = "console=ttyATH0,250000";
+	};
+
+	ahb {
+		apb {
+			pinmux_extended: pinmux@18040030 {
+				compatible = "pinctrl-single";
+				reg = <0x18040030 0x4>;
+
+				pinctrl-single,bit-per-mux;
+				pinctrl-single,register-width = <32>;
+				pinctrl-single,function-mask = <0x1>;
+				#pinctrl-cells = <2>;
+
+				enable_gpio11: pinmux_enable_gpio11 {
+					pinctrl-single,bits = <0x0 0x200 0x200>;
+				};
+			};
+
+			pinmux_bootstrap: pinmux@180600ac {
+				compatible = "pinctrl-single";
+				reg = <0x180600ac 0x4>;
+
+				pinctrl-single,bit-per-mux;
+				pinctrl-single,register-width = <32>;
+				pinctrl-single,function-mask = <0x1>;
+				#pinctrl-cells = <2>;
+
+				enable_gpio26_gpio27: pinmux_enable_gpio26_gpio27 {
+					pinctrl-single,bits = <0x0 0x40000 0x40000>;
+				};
+			};
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		wlan {
+			label = "yun:blue:wlan";
+			gpios = <&gpio 0 GPIO_ACTIVE_HIGH>;
+			linux,default-trigger = "phy0tpt";
+		};
+
+		usb {
+			label = "yun:white:usb";
+			gpios = <&gpio 1 GPIO_ACTIVE_HIGH>;
+			trigger-sources = <&hub_port1>;
+			linux,default-trigger = "usbport";
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		config {
+			label = "config";
```

Comment:
WPS?

Reply:
I haven't decided on it yet. Maybe KEY_WPS_BUTTON is not a good choice here (nor in ar71xx). Who would use WPS on a tiny dev board?
The label on the board reads "WLAN_RST", and the stock FW uses it for network reset (5 sec) and factory reset (30 sec).
KEY_RESTART is not a good choice either. The button is easily accessible, any accidental touch might cause reboot. 
I'm thinking of using another code like BTN_0 or KEY_CONFIG. Then users can choose what the button should do by writing their own script or by creating a link to an existing script file (in /etc/rc.button).